### PR TITLE
[OP] Add `rms_norm` into TOPI

### PR DESCRIPTION
**Pull request by cyx-6**
This PR introduces the operator root mean square, `rms_norm`, into TOPI.

### CODE REVIEW

File: include/tvm/topi/nn/rms_norm.h [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * \param data N-D tensor with shape [d_0, d_1, ..., d_{N-1}]
+ * \param weight K-D tensor with shape [r_0, r_1, ..., r_{K-1}] where K == len(axis) and
+ *               d_{axis_k} == r_k
+ * \param axis The axis to normalize over.
+ * \param epsilon The epsilon value to avoid division by zero.
+ * \param name The name of the operation.
+ * \param tag The tag to mark the operation.
+ * \return The normalized tensor, with the same shape as data.
+ */
+inline Tensor rms_norm(const Tensor& data, const Tensor& weight, const Array<Integer>& axis,
```
Comment (junrushao) [inline include/tvm/topi/nn/rms_norm.h:50]: NOTE: the `bias` term is missing

File: tests/python/topi/python/test_topi_rms_norm.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+_rms_norm_schedule = {
+    "generic": topi.generic.schedule_injective,
+}
+
+
+# only test on llvm because schedule is missing
+@tvm.testing.parametrize_targets("llvm")
+@pytest.mark.parametrize("shape,axis", [([4, 16], (1,)), ([4, 16, 16], (1, 2))])
```
Comment (sunggg) [inline tests/python/topi/python/test_topi_rms_norm.py:36]: If it also support symbolic shape, would you add a testcase?

File: include/tvm/topi/nn/rms_norm.h [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                       double epsilon, std::string name = "T_rms_norm",
+                       std::string tag = kInjective) {
+  const auto& data_type = data->dtype;
+  const auto& weight_type = weight.defined() ? weight->dtype : data_type;
+  ICHECK(data_type == weight_type) << "rms_norm: data and weight must have the same type";
+  ICHECK(data_type == DataType::Float(32) || data_type == DataType::Float(16))
+      << "rms_norm: only support float32 and float16 for now";
+  bool is_float16 = data_type == DataType::Float(16);
+
+  auto x = is_float16 ? cast(data, DataType::Float(32)) : data;
```
Comment (sunggg) [inline include/tvm/topi/nn/rms_norm.h:60]: Why do we need casting here?

**Review by sunggg — COMMENTED** [review]
Thank you for the nice work! A few minor comments.

File: include/tvm/topi/nn/rms_norm.h (hunk 1 above)
Comment (cyx-6) [reply include/tvm/topi/nn/rms_norm.h]: The `bias` term has been added.

File: include/tvm/topi/nn/rms_norm.h (hunk 3 above)
Comment (cyx-6) [reply include/tvm/topi/nn/rms_norm.h]: Actually no, and all the `cast` have been removed.

File: tests/python/topi/python/test_topi_rms_norm.py (hunk 2 above)
Comment (cyx-6) [reply tests/python/topi/python/test_topi_rms_norm.py]: yes, the symbolic shape test has been added.

**Review by MasterJH5574 — APPROVED** [review]

**Pull request merged**